frontend/ui/src/pages/Inbox/ConversationListItem/index.tsx

```diff
@@ -93,21 +93,23 @@ const ConversationListItem = (props: ConversationListItemProps) => {
 
     if (typeof lastMessageContent === 'string') {
       if (lastMessageContent.includes('&Body=' && '&FromCountry=')) {
-        const startText = lastMessageContent.search('&Body=');
-        const endText = lastMessageContent.search('&FromCountry=');
-        const textLength = endText - startText;
-        const enCodedText = lastMessageContent.substring(startText + 6, startText + textLength);
-        const replaced = enCodedText.split('+').join(' ');
-        const text = decodeURIComponent(replaced);
-        return text;
+        return (
+          <SourceMessage
+            source={conversation.channel.source}
+            contentType="message"
+            content={conversation.lastMessage}
+            isTwilioConversationListItem={true}
+          />
+        );
       } else if (lastMessageContent.includes('&Body=' && '&To=whatsapp')) {
-        const startText = lastMessageContent.search('&Body=');
-        const endText = lastMessageContent.search('&To=whatsapp');
-        const textLength = endText - startText;
-        const enCodedText = lastMessageContent.substring(startText + 6, startText + textLength);
-        const replaced = enCodedText.split('+').join(' ');
-        const text = decodeURIComponent(replaced);
-        return text;
+        return (
+          <SourceMessage
+            source={conversation.channel.source}
+            contentType="message"
+            content={conversation.lastMessage}
+            isTwilioConversationListItem={true}
```

Comment:
The API of the render library needs to be source agnostic. What does this attribute change?

Reply:
the styling of the message itself, so it not a bubble as in messageList @chrismatix

Reply:
I agree with @chrismatix the render lib should be agnostic. That's what I meant in my previous comment. Either we don't use the render lib for this and we do it outside checking all the types, or have a different component in the render lib for the preview of the message. But we shouldn't change the `SourceMessage` for this. @airyhq/codeowners-frontend